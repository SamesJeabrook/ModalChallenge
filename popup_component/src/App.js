import React from 'react';
import PopupModal from './popupModal/popupParent'

function App() {
  return (
    <div>
       <PopupModal />
    </div>
  );
}

export default App;
