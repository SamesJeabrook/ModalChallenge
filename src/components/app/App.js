import React from "react";
import "./app.less";
import Modal from "../modal/Modal";
import SampleMessage from "../SampleMessage";
import SuccessMessage from "../successmessage/SuccessMessage";

export default class App extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      isDefaultModalOpen: false,
      isSuccessModalOpen: false,
      onCloseMessage: ""
    };
    this.closeModal = this.closeModal.bind(this);
    this.reset = this.reset.bind(this);
  }

  openModal(modalName) {
    const formattedModalName = `is${modalName}ModalOpen`;
    this.setState({
      [formattedModalName]: true
    });
  }

  reset() {
    this.setState({
      onCloseMessage: ""
    });
  }

  closeModal(values, modalName) {
    const formattedModalName = `is${modalName}ModalOpen`;
    this.setState({
      [formattedModalName]: false,
      onCloseMessage: values.message
    });
    if(values.message !== ''){
        setTimeout(this.reset, 5000);
    }
  }

  render() {
    const {
      isDefaultModalOpen,
      isSuccessModalOpen,
      onCloseMessage
    } = this.state;
    const SUCCESS_MODAL = "Success";
    const DEFAULT_MODAL = "Default";
    const hasCloseMessage = onCloseMessage !== "";
    return (
      <div>
        <header className="header">
          <h1 className="header-title">Modal demos</h1>
        </header>
        <div className="content">
          {
           isDefaultModalOpen &&      
          <Modal
            isModalOpen={isDefaultModalOpen}
            modalSize="larges"
            modalName={DEFAULT_MODAL}
            modalTitle="Welcome !"
            closeButtonLabel="Close"
            onClose={this.closeModal}
            canPassDataOnClose={true}
            children={<SampleMessage />}
          />
        }

        {isSuccessModalOpen && 
          <Modal
            isModalOpen={isSuccessModalOpen}
            modalName={SUCCESS_MODAL}
            modalSize="small"
            modalTitle="Success !"
            closeButtonLabel="Close"
            canPassDataOnClose={false}
            onClose={this.closeModal}
            autoClose={true}
            children={<SuccessMessage />}
          />
        }

          <div className="action-buttons-wrapper">
            <button
              className="button button_primary"
              onClick={e => this.openModal(DEFAULT_MODAL)}
            >
              Click me for modal
            </button>
            <button
              className="button button_success"
              onClick={e => this.openModal(SUCCESS_MODAL)}
            >
              Click me for Success modal
            </button>
          </div>

          {hasCloseMessage && (
            <p className={`note ${!hasCloseMessage ? "hide" : ""}`}>
              {onCloseMessage}
            </p>
          )}
          
        </div>
      </div>
    );
  }
}
