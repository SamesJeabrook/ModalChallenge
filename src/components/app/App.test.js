import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import {shallow} from 'enzyme';

test('Validate App component', function(){
    // arrange
    const html = (<App/>);
    const tree = renderer.create(html).toJSON();

    expect(tree).toMatchSnapshot();   
})

test('It opens the modal on click of button', function(){

    const appComponent = shallow(<App />)
    const modalButton = appComponent.find('.button_primary');
    modalButton.simulate('click');

    expect(appComponent.state().isDefaultModalOpen).toBeTruthy();
    
})

test('It opens the modal and make sure only one modal opens', function(){

    const appComponent = shallow(<App />)
    const modalButton = appComponent.find('.button_primary');
    modalButton.simulate('click');

    expect(appComponent.state().isDefaultModalOpen).toBeTruthy();
    expect(appComponent.state().isSuccessModalOpen).toBeFalsy();

})

test('It shows the close message, if receive from modal', function(){

    const appComponent = shallow(<App />)
    const modalButton = appComponent.find('.button_primary');
    modalButton.simulate('click');

    appComponent.setState({onCloseMessage: 'Close'});

    expect(appComponent.find('.note').hasClass('hide')).toBeFalsy();

})

test('It does not show the close message, if not received from modal', function(){

    const appComponent = shallow(<App />)
    const modalButton = appComponent.find('.button_primary');
    modalButton.simulate('click');

    expect(appComponent.find('.note').exists()).toBeFalsy();

})