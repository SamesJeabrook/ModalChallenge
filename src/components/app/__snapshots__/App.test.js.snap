// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate App component 1`] = `
<div>
  <header
    className="header"
  >
    <h1
      className="header-title"
    >
      Modal demos
    </h1>
  </header>
  <div
    className="content"
  >
    <div
      className="action-buttons-wrapper"
    >
      <button
        className="button button_primary"
        onClick={[Function]}
      >
        Click me for modal
      </button>
      <button
        className="button button_success"
        onClick={[Function]}
      >
        Click me for Success modal
      </button>
    </div>
  </div>
</div>
`;
